src/LessMsi.Gui/MainForm.cs

```diff
@@ -1034,22 +1036,20 @@ private void searchFileToolStripMenuItem_Click(object sender, EventArgs e)
 	        if (IsFileTabSelected)
 	        {
 		        searchPanel.SearchDataGrid(this.fileGrid,
-		                                   (o, args) => Presenter.BeginSearching(args.SearchString),
-		                                   (o, args) => { Presenter.BeginSearching(""); }
+										   Presenter.ExecuteFileSearch,
+										   () => { Presenter.ExecuteFileSearch(this.fileGrid, string.Empty); }

```

Comment:
Would you like me to keep it as events in this case, as to minimize the amount of changes needed?

Reply:
And some events still need to be de-registered, for example:
the  search controls registers itself to the resize event of a tab page, so at least this one needs to be unregistered on dispose.

Reply:
No concern about these callback events. 

Regarding need to unsubscribe: I remember when I wrote this thinking that calling `Dispose` unsubscribes, but I think you're right and it won't do that (it was late :)).  Given the need to remove `Parent_Resize` maybe your approach is cleaner after all. I'm fine either way. Please just do some testing and you decide. Thanks again for walking me through this! This UI code all makes me nervous because it doesn't have good automated tests so please bare with me on my questions!

Reply:
Yeah, sadly it is very hard to reliably test UI.

The dispose would only add value in an instance like this if we add an OnDispose event handler, and actively unsubscribe in that.
In this particular instance, it does make the code slightly more obvious when doing that, and re-adding an assert back in the search function that the last search grid is actually the current one.

Another thing that could be an option is to create one search dialog per tab.
Then a search could even be remembered when switching back to a tab.

Reply:
I think "remembering" the search between tabs is going to be more confusing to the user. Better to clear it probably (the app might remember, but the user won't :)). Again, as for dispose or not, you decide. Just please test and make sure it works good and I'll trust your judgement.